docs/reference/src/language/syntax/types.md

```diff
@@ -87,6 +96,13 @@ All colors and brushes define the following methods:
     Returns a new color that is a mix of this color and `other`, with a proportion
     factor given by \a factor (which will be clamped to be between `0.0` and `1.0`).
 
+- **`linear-mix(other: brush, factor: float) -> brush`**
+
+    Returns a new color that is a mix of this color and `other`, with a proportion
+    factor given by \a factor (which will be clamped to be between `0.0` and `1.0`).
+    This is different to the `mix()` method above in that it works directly on ARGB values instead
+    of converting them to HSVA first.
```

Comment:
Could you elaborate what's the use case for this function compared to mix? In which case would you prefer linear-mix over mix

Reply:
When i make a gradient of RGB colours and use a slider to select which two colours to mix, the `mix()` method makes it "step" between successive mixes and it looks bad. The linear method is much much smoother.

From what I see, it's the `channels_factor` that is causing this in mix(). I'll try a few things.

Reply:
i'd like to see an example of what you're talking about.

Reply:
Do you mean adding an example in code? It's pretty simple - if I change [https://gitlab.com/asus-linux/asusctl/-/blob/wip/keyboard_power/rog-control-center/ui/widgets/colour_picker.slint?ref_type=heads#L72] from `interpolate` to `mix` I get a *jump* when the slider switches to the next colour.